md_main.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+import logging
+import pandas as pd
+import pathlib
+import shutil
+import sqlite3
+from os import path
+
+from scripts.database_operations import *
+
+# todo: turn all print and error statements into logging
+# test run: python md_main.py metadata -m ./test/data/gordon_db_ready.tsv --db "test.db"
+
+EXPECTED_COLS_MD = [
+    "external_source",
+    "external_accession",
+    "filename",
+    "filepath",
+    "bioproject",
+    "biosample",
+    "wgs_master",
+    "refseq_category",
+    "taxid",
+    "species_taxid",
+    "organism_name",
+    "infraspecific_name",
+    "isolate",
+    "version_status",
+    "assembly_level",
+    "release_type",
+    "genome_rep", 
+    "seq_rel_date",
+    "asm_name",
+    "submitter",
+    "gbrs_paired_asm",
+    "paired_asm_comp",
+    "ftp_path",
+    "excluded_from_refseq",
+    "relation_to_type_material",
+    "asm_not_live_date"
+    ]
+
+INSERT_COLS_IDENTIFIER = [
+    'filename',
+    'filepath',
+    'external_accession',
+    'external_source'
+]
+
+INSERT_COLS_MD = EXPECTED_COLS_MD.copy()
+INSERT_COLS_MD.remove('filename')
+INSERT_COLS_MD.remove('filepath')
+
+def check_input(path_md):
+    """Checks if the input md file is valid and returns metadata as a pandas dataframe."""
+    if path.isfile(path_md):
+        md = pd.read_csv(path_md, sep='\t')
+        actual_cols = md.columns.to_list()
+        print(actual_cols)
+        actual_cols_n, expected_cols_n = len(actual_cols), len(EXPECTED_COLS_MD)
+        if actual_cols_n == expected_cols_n:
+            for i in range(actual_cols_n):
+                if actual_cols[i] != EXPECTED_COLS_MD[i]:
+                    raise IOError(f'Metadata does not contain expected headers. Check {actual_headers[i]}')
+        else:
+            raise IOError(f'Metadata does not have {expected_cols_n} columns.')
+        print('Metadata loaded')
+        return md
+    else: 
+        raise FileNotFoundError(path_md)
+
+def insert_into_identifier(conn, md):
+    """Insert relevant columns into identifiers and return corresponding entity ids."""
+    cursor = conn.cursor()
+    _fileinfo = md[INSERT_COLS_IDENTIFIER]
+
+    # get existing entity_id
+    cursor.execute("SELECT MAX(entity_id) FROM identifier")
+    max_existing_id = cursor.fetchone()[0]
+    if max_existing_id is None:
+        max_existing_id = 0
+
+    # insert new records into identifier
+    _fileinfo.to_sql('identifier', conn, if_exists='append', index=False)
+
+    # extract entity ids of newly added records
+    cursor.execute(f"SELECT entity_id FROM identifier WHERE entity_id > {max_existing_id}")
```

Comment:
This is not assured to work correctly if there are multiple connections to the database. putting this all in a transaction would be one way to do it, or rewriting the `.to_sql` call so that it uses [`RETURNING`](https://www.sqlite.org/lang_returning.html)